### ISSUE TITLE
Is it possible to have both "OnTeamsChannelCreatedAsync" and "OnConversationUpdateActivityAsync" events ?

### ISSUE THREAD

**Issue by piyumi25** [MicrosoftDocs/msteams-docs#2365]
I was trying to create a teams bot application to test basic events by following this documentation https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/subscribe-to-conversation-events?tabs=dotnet#team-renamed. I experienced that if I am having both "OnTeamsChannelCreatedAsync" and "OnConversationUpdateActivityAsync" events and I created a channel then only "OnConversationUpdateActivityAsync" is fired. If i remove the "OnConversationUpdateActivityAsync" event then "OnTeamsChannelCreatedAsync " is fired. Is that the expected behavior? 
```
[Obsolete]
        protected override async Task OnConversationUpdateActivityAsync(ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
        {
            AdaptiveCard card = new AdaptiveCard();
            //card.Speak = "<s>Your  meeting about \"Adaptive Card design session\"<break strength='weak'/> is starting at 12:30pm</s><s>Do you want to snooze <break strength='weak'/> or do you want to send a late notification to the attendees?</s>";

            card.Body.Add(new AdaptiveTextBlock()
            {
                Text = $"Welcome to search archive bot",
                Size = AdaptiveTextSize.Large,
                Weight = AdaptiveTextWeight.Bolder
            });
            card.Body.Add(new AdaptiveTextBlock()
            {
                Text = $"Search for an email here...",
                IsSubtle = true,
                Size = AdaptiveTextSize.Medium,
                Weight = AdaptiveTextWeight.Default
            });
            card.Body.Add(new AdaptiveTextInput()
            {
                Id = "SearchText",
                Placeholder = "search"
            });
            card.Actions.Add(new AdaptiveSubmitAction()
            {
                Type = AdaptiveSubmitAction.TypeName,
                Title = "Search",
                Data = new JObject { { "Type", "search" } },
            });
            Attachment attachment = new Attachment()
            {
                ContentType = AdaptiveCard.ContentType,
                Content = card
            };
            await turnContext.SendActivityAsync(MessageFactory.Attachment(attachment));
        }
        protected virtual async Task OnTeamsMembersAddedAsync(IList<ChannelAccount> membersAdded, TeamInfo teamInfo, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
        {
            foreach (TeamsChannelAccount member in membersAdded)
            {
                if (member.Id == turnContext.Activity.Recipient.Id)
                {
                    // Send a message to introduce the bot to the team
                    var heroCard = new HeroCard(text: $"The {member.Name} bot has joined {teamInfo.Name}");
                    await turnContext.SendActivityAsync(MessageFactory.Attachment(heroCard.ToAttachment()), cancellationToken);
                }
                else
                {
                    var heroCard = new HeroCard(text: $"{member.Name} joined {teamInfo.Name}");
                    await turnContext.SendActivityAsync(MessageFactory.Attachment(heroCard.ToAttachment()), cancellationToken);
                }
            }
        }
        protected virtual async Task OnTeamsMembersRemovedAsync(IList<ChannelAccount> membersRemoved, TeamInfo teamInfo, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
        {
            foreach (TeamsChannelAccount member in membersRemoved)
            {
                if (member.Id == turnContext.Activity.Recipient.Id)
                {
                    // The bot was removed
                    // You should clear any cached data you have for this team
                }
                else
                {
                    var heroCard = new HeroCard(text: $"{member.Name} was removed from {teamInfo.Name}");
                    await turnContext.SendActivityAsync(MessageFactory.Attachment(heroCard.ToAttachment()), cancellationToken);
                }
            }
        }
        protected override async Task OnTeamsTeamRenamedAsync(TeamInfo teamInfo, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
        {
            var heroCard = new HeroCard(text: $"{teamInfo.Name} is the new Team name");
            await turnContext.SendActivityAsync(MessageFactory.Attachment(heroCard.ToAttachment()), cancellationToken);
        }
        protected override async Task OnTeamsChannelCreatedAsync(ChannelInfo channelInfo, TeamInfo teamInfo, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
        {
            var heroCard = new HeroCard(text: $"{channelInfo.Name} is the Channel created");
            await turnContext.SendActivityAsync(MessageFactory.Attachment(heroCard.ToAttachment()), cancellationToken);
        }
        protected override async Task OnTeamsChannelRenamedAsync(ChannelInfo channelInfo, TeamInfo teamInfo, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
        {
            var heroCard = new HeroCard(text: $"{channelInfo.Name} is the new Channel name");
            await turnContext.SendActivityAsync(MessageFactory.Attachment(heroCard.ToAttachment()), cancellationToken);
        }
        protected override async Task OnTeamsChannelDeletedAsync(ChannelInfo channelInfo, TeamInfo teamInfo, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
        {
            var heroCard = new HeroCard(text: $"{channelInfo.Name} is the Channel deleted");
            await turnContext.SendActivityAsync(MessageFactory.Attachment(heroCard.ToAttachment()), cancellationToken);
        }
        protected override async Task OnReactionsAddedAsync(IList<MessageReaction> messageReactions, ITurnContext<IMessageReactionActivity> turnContext, CancellationToken cancellationToken)
        {
            foreach (var reaction in messageReactions)
            {
                var newReaction = $"You reacted with '{reaction.Type}' to the following message: '{turnContext.Activity.ReplyToId}'";
                var replyActivity = MessageFactory.Text(newReaction);
                var resourceResponse = await turnContext.SendActivityAsync(replyActivity, cancellationToken);
            }
        }
        protected override async Task OnReactionsRemovedAsync(IList<MessageReaction> messageReactions, ITurnContext<IMessageReactionActivity> turnContext, CancellationToken cancellationToken)
        {
            foreach (var reaction in messageReactions)
            {
                var newReaction = $"You removed the reaction '{reaction.Type}' from the following message: '{turnContext.Activity.ReplyToId}'";
                var replyActivity = MessageFactory.Text(newReaction);
                var resourceResponse = await turnContext.SendActivityAsync(replyActivity, cancellationToken);
            }
        }
```

[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0a6d00a4-dd28-33c4-002e-2e8e7c653ce2
* Version Independent ID: 0856927a-25f4-b1e0-5f1b-d55dc61796f3
* Content: [Subscribe to conversation events - Teams](https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/subscribe-to-conversation-events?tabs=dotnet)
* Content Source: [msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md](https://github.com/MicrosoftDocs/msteams-docs/blob/master/msteams-platform/bots/how-to/conversations/subscribe-to-conversation-events.md)
* Product: **msteams**
* GitHub Login: @WashingtonKayaker
* Microsoft Alias: **anclear**

**Comment by RamaMohanaChary** [comment]
Hi @piyumi25,

Sorry for delay response, we can able to reproduce the issue, checking internally, will get back to you with in short time.

**Comment by piyumi25** [comment]
Thanks @RamaMohanaChary

**Comment by RamaMohanaChary** [comment]
Hi @piyumi25,

We are able to reproduce the issue, assigned to particular person. We don't have any ETA. For now we are closing this question.
Thank you for reaching us.

**Comment by clearab** [comment]
Yes, that is the expected behavior. The OnConversationUpdateActivityAsync method dispatches to multiple other more specific events, if you override it none of those more specific events will fire. You can see the implementation of this method here: https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs#L398.

Normally, you'll only override one of the more specific events, like channel creation or member added. If you need to override the OnConversationUpdate method, you'll need to make sure you handle whichever specific case you're looking for, and then call the base method so that it can finish dispatching to the appropriate events.

**Comment by piyumi25** [comment]
Thanks for your attention to the issue

**Comment by clearab** [comment]
Its actually a bit weirder than I mentioned, since the implementation of OnConversationUpdate in the TeamsActivityHandler is actually overriding the virtual method in the base ActivityHandler, and is not itself a virtual method. So what you're overriding is actually the base implementation, and completely avoiding the TeamsActivityHandler version, so you'd actually need to re-implement the switch statement in the TeamsActivityHandler class if you really need to override this method.

tl;dr - don't override this method and instead override one of the more specific methods it dispatches to if at all possible
